js/tips-tricks.md

```diff
@@ -0,0 +1,26 @@
+# JavaScript / Tips & Tricks
+
+#### Singleton classes
+
+```js
+class Foo {
+    // ...
+}
+
+class Singleton {
+   static instance = null;
+
+   static getInstance() {
+      if(!this.instance) {
+         this.instance = new Foo();
+      }
+
+      return this.instance;
+   }
+}
+
+const s1 = Singleton.getInstance();
+const s2 = Singleton.getInstance();
+
+console.log(s1 === s2); // => true
```

Comment:
IMO this and the instantiations are not needed, should be obvious.

Reply:
Makes sense, thanks! After all that's the gist of singleton classes.

Reply:
https://github.com/c-hive/guides/pull/43/commits/8fed5a0eedb9b72041c7049afcd2709a5f4354a6